### PR TITLE
Win32 MessageBox sample

### PR DESCRIPTION
**Pull request by domesticmouse**
Proving out how to use Win32 UTF16 API from Dart.

### CODE REVIEW

**Comment by mit-mit** [comment]
Overall looks good to me, cc @dcharkes for technical review

**Comment by domesticmouse** [comment]
PTAL @dcharkes

File: ffi/system-command/win32ui.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi' as ffi;
```
Comment (dcharkes) [inline ffi/system-command/win32ui.dart]: We don't really use `as ffi` in any of the samples inside the SDK anymore.

File: ffi/system-command/win32ui.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  // Look up the `MessageBoxW` function.
+  final MessageBoxP =
+      user32.lookupFunction<MessageBoxC, MessageBoxDart>('MessageBoxW');
+
+  // Allocate pointers to Utf16 arrays containing the command arguments.
+  final messageP = Utf16.toUtf16(message);
+  final captionP = Utf16.toUtf16(caption);
+
+  // Invoke the command, and free the pointers.
+  var result =
```
Comment (dcharkes) [inline ffi/system-command/win32ui.dart]: final?

Comment (domesticmouse) [reply ffi/system-command/win32ui.dart]: done

File: ffi/system-command/win32ui.dart (hunk 1 above)
Comment (domesticmouse) [reply ffi/system-command/win32ui.dart]: dropped the `as ffi`.

**Pull request merged**